Source/NSStoryboard.m

```diff
@@ -0,0 +1,276 @@
+                                           creator: block];
+}
+
+- (id) instantiateControllerWithIdentifier: (NSStoryboardSceneIdentifier)identifier
+{
+  return [self instantiateControllerWithIdentifier: identifier
+                                           creator: nil];
+}
+
+- (id) instantiateControllerWithIdentifier: (NSStoryboardSceneIdentifier)identifier
+                                   creator: (NSStoryboardControllerCreator)block
+{
+  id result = nil;
+  NSMutableArray *topLevelObjects = [NSMutableArray arrayWithCapacity: 5];
+  NSDictionary *table = [NSDictionary dictionaryWithObjectsAndKeys: topLevelObjects,
+                                      NSNibTopLevelObjects,
+                                      NSApp,
+                                      NSNibOwner,
+                                      nil];
+  GSModelLoader *loader = [GSModelLoaderFactory modelLoaderForFileType: @"xib"];
+  BOOL  success = [loader loadModelData: [_transform dataForIdentifier: identifier]
+                      externalNameTable: table
+                               withZone: [self zone]];
+    
+  if (success)
+    {
+      NSMutableArray *seguesToPerform = [NSMutableArray array];
+      NSMutableArray *placeholders = [NSMutableArray array];
+      NSMapTable *segueMap = [_transform segueMapForIdentifier: identifier];
+      NSWindowController *wc = nil;
+      NSViewController *vc = nil;
+      NSWindow *w = nil;
+            
+      FOR_IN(id, o, topLevelObjects)
+        {
+          if ([o isKindOfClass: [NSWindowController class]])
+            {
+              wc = (NSWindowController *)o;
+              [wc _setSegueMap: segueMap];
+              [wc _setTopLevelObjects: topLevelObjects];
+              [wc _setStoryboard: self];
+              [wc _setOwner: NSApp];
+              result = o;
+            }
+          else if ([o isKindOfClass: [NSViewController class]])
+            {
+              vc = (NSViewController *)o;
+              [vc _setSegueMap: segueMap];
+              [vc _setTopLevelObjects: topLevelObjects];
+              [vc _setStoryboard: self];
+              result = o;
+            }
+          else if ([o isKindOfClass: [NSWindow class]])
+            {
+              w = (NSWindow *)o;
+            }
+          else if ([o isKindOfClass: [NSControllerPlaceholder class]])
+            {
+              [placeholders addObject: o];
+            }             
+        }
+      END_FOR_IN(topLevelObjects);
+      
+      // Process action proxies after so we know we have the windowController...
+      FOR_IN(id, o, topLevelObjects)
+        {
+          if ([o isKindOfClass: [NSStoryboardSeguePerformAction class]])
+            {
+              NSStoryboardSeguePerformAction *ssa = (NSStoryboardSeguePerformAction *)o;
+              [ssa setSender: result]; // resolve controller here...
+              [ssa setIdentifierToSegueMap: [_transform identifierToSegueMap]];
```

Comment:
As far as I understand this map exists only once in a storyboard. Why not keep it on the storyboard itself instead of the action?

Reply:
it's actually calculated per controller.   So, while it could exist on the storyboard, it's easier to resolve it if it's on the action.

Reply:
If this was your intention then the actual code is wrong. There is only one such mach and with in the NSStoryboardSeguePerformAction you only ever use the key "application".
Not sure what to believe, the code or your explanation.